### DISCUSSION TITLE
Merkle contract for Lockup Dynamic

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/airdrops#30] [Ideas]
Support for Lockup Dynamics inside the Merke suite of contracts has been requested by one of our users. They're looking to define a custom shape, so the Linear flavor doesn't match their proposed design.

When/how/should we start working on `SablierV2MerkleStreamerFactoryLD`?

Worth mentioning this user is considering a very big distribution, so there's a considerable upside in Sablier being able to support their campaign.

**Comment by andreivladbrg** [comment]
> When/how/should we start working on SablierV2MerkleStreamerFactoryLD?

I think it makes the most sense to implement this contract very early next year, ideally in the first two weeks of the year. So that we can get it audited fast.

**Comment by PaulRBerg** [comment]
> When/how/should we start

Early next year.

>  working on `SablierV2MerkleStreamerFactoryLD`?

Just `src/SablierV2MerkleStreamerLD`. We don't need another factory.

---

Created an issue for it:

https://github.com/sablier-labs/v2-periphery/issues/232

**Reply by razgraf** [reply]
You're correct. No new factory "flavor", we can use the same point of entry - just the factory itself has to be updated to contain a factory method for the LD Merkle Streamer.

**Comment by PaulRBerg** [comment]
Since Lockup v2.0, the only shapes for which LD is needed are `Exponential` and `Exponential Cliff`.

We haven't seen much demand for those curves, so I suggest closing this discussion.

WDYT @sablier-labs/engineers?

**Reply by smol-ninja** [reply]
Completely agree with closing this discussion for now.
